src/components/search-input.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './search-input.css';
+
+const ENTER_KEY_CODE = 13;
+
+class SearchInput extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: ''
+		};
+		this.search = this.search.bind(this);
+		this.handleOnClick = this.handleOnClick.bind(this);
+		this.handleOnChange = this.handleOnChange.bind(this);
+		this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+	}
+
+	handleOnChange(e) {
+		this.setState({
+			value: e.target.value
+		});
+	}
+
+	handleOnKeyDown(e) {
+		if (e.keyCode === ENTER_KEY_CODE) {
+			this.search();
+		}
+	}
+
+	handleOnClick() {
```

Comment:
It should be either `handle... ` or `on...`. The convention is that `on...` is used inside stateless rendering components to pass handlers into HTML elements, and `handle...` is used in stateful components that react on the respective events.